acurl/acurl/utils.py

```diff
@@ -0,0 +1,54 @@
+from collections import defaultdict
+
+
+class CaseInsensitiveDefaultDict(defaultdict):
+    # Base on https://stackoverflow.com/a/32888599 but tweaked for python3
+    # and using defaultdict as a base
+    @classmethod
+    def _k(cls, key):
+        return key.lower() if isinstance(key, str) else key
+
+    def __init__(self, *args, **kwargs):
+        super(CaseInsensitiveDefaultDict, self).__init__(*args, **kwargs)
+        self._convert_keys()
+
+    def __getitem__(self, key):
+        return super(CaseInsensitiveDefaultDict, self).__getitem__(self.__class__._k(key))
+
+    def __setitem__(self, key, value):
+        super(CaseInsensitiveDefaultDict, self).__setitem__(self.__class__._k(key), value)
+
+    def __delitem__(self, key):
+        return super(CaseInsensitiveDefaultDict, self).__delitem__(self.__class__._k(key))
+
+    def __contains__(self, key):
+        return super(CaseInsensitiveDefaultDict, self).__contains__(
+            self.__class__._k(key)
+        )
+
+    def has_key(self, key):
+        return super(CaseInsensitiveDefaultDict, self).has_key(self.__class__._k(key))
```

Comment:
I'm a bit confused about this one because it's not an api method for dicts in python...

Reply:
Looks like it was removed some time between 2 and 3. I didn't even know it had gone.

Reply:
I don't even know what python 2 is any more 😝

Reply:
Yeah I just read that, it will be expunged from my next commit.